scripts/globals/bluemagic.lua

```diff
@@ -46,6 +46,29 @@ local function calculateWSC(attacker, params)
     return wsc
 end
 
+local function calculateDEXvsAGICritRate(attacker, target)
```

Comment:
Should move this into https://github.com/LandSandBoat/server/blob/base/scripts/globals/combat/physical_utilities.lua

Reply:
IF thats the case, I have a request:
```lua
xi.combat.physical.calculateMeleeCriticalRate = function(actor, target)
    -- See reference at https://www.bg-wiki.com/ffxi/Critical_Hit_Rate
    local criticalRate = 0
    local dDex         = actor:getStat(xi.mod.DEX) - target:getStat(xi.mod.AGI)

    if dDex > 50 then
        criticalRate = 0.15
    elseif dDex >= 40 then
        criticalRate = (dDex - 35) / 100
    elseif dDex >= 30 then
        criticalRate = 0.04
    elseif dDex >= 20 then
        criticalRate = 0.03
    elseif dDex >= 14 then
        criticalRate = 0.02
    elseif dDex >= 7 then
        criticalRate = 0.01
    end

    -- We should seriously consider adding merits, job points and other modifiers here. 
    
    return criticalRate
end
```

Reasoning:
- This files are the base of weaponskills + other actions being rewriten to be... less chaotic.
- Variable naming: I personally prefer to name variables after what wikis name them. Same terms = more clarity. I also dislike abreviations.
- Order: Critical hit rate is already defined as 0, so theres no need for a step setting it to 0.
- We should also really consider puting mods that affect critical rate in there aswell. Because theres no need to add them in a different function, nor independently in blue magic, weaponskills, TP moves, JAs, etc

Reply:
Those are my dominions. Feel spooked. UHHHhHhHHhH....

Reply:
Done and done.

Added a third param that if passed as true, bypasses the other modifiers so i could change weaponskills.lua as little as possible (and for flexibility)